src/core/SampleBuffer.cpp

```diff
@@ -1017,30 +1017,63 @@ void SampleBuffer::visualize(
 	const float ySpace = h * 0.5f;
 	const int nbFrames = focusOnRange ? toFrame - fromFrame : m_frames;
 
-	const int fpp = qBound<int>(1, nbFrames / w, 20);
-	QPointF * l = new QPointF[nbFrames / fpp + 1];
-	QPointF * r = new QPointF[nbFrames / fpp + 1];
+	const int fpp = qBound<int>(1, nbFrames / w, 1000);
+	QPointF * fMax = new QPointF[nbFrames / fpp * 2 + 2];
+	QPointF * fRms = new QPointF[nbFrames / fpp * 2 + 2];
```

Comment:
You forgot to delete fRms at the end of the function. I'd suggested to use vectors or add the deleter.
```suggestion
	std::vector<QPointF> fMax(nbFrames / fpp * 2 + 2);
	std::vector<QPointF> fRms(nbFrames / fpp * 2 + 2);
```